### PR TITLE
Update SoapMessageEventSubscriber.php

### PR DESCRIPTION
**Pull request by Seriyyy95**
Missing prefix in root namespace

### CODE REVIEW

**Comment by proggeler** [comment]
@Seriyyy95 Thanks for the PR (I was not aware this was changed in jms-serializer)
But can you add some test cases for the parts you have changed? 
(also can you please apply the [PSR-12](https://www.php-fig.org/psr/psr-12/) coding standard)

**Comment by Seriyyy95** [comment]
Done.

File: tests/SoapSerializationTest.php [hunk 1]
```diff
@@ -38,6 +39,36 @@ public function testSerialization(string $name, int $complexity, DateTime $date)
...
     }
 
+    /**
+     * @dataProvider provideLanguage
+     *
+     * @param string $name
+     * @param int $complexity
+     * @param DateTime $date
+     */
+    public function testSerializationWithPrefix(string $name, int $complexity, DateTime $date)
```
Comment (proggeler) [inline tests/SoapSerializationTest.php:49]: This tests the registered namespace and prefix, not is the root node for the message actual contains that prefix.
I think you should use DOMDocument here to ensure the prefix is set on the expected element(s)

(see SoapHeaderEventSubscriberTest.php)

File: src/SoapHeaderEventSubscriber.php [hunk 2]
```diff
@@ -69,9 +69,9 @@ public function addSoapHeader(ObjectEvent $event)
                 )
             );
             
-            if($metadata->xmlRootPrefix !== null){
+            if ($metadata->xmlRootPrefix !== null) {
                 $tagName = $metadata->xmlRootPrefix . ':' . $metadata->xmlRootName;
-            }else{
+            } else {
```
Comment (proggeler) [inline src/SoapHeaderEventSubscriber.php:74]: You could loose the else here
```
            $tagName = $metadata->xmlRootName;
            if ($metadata->xmlRootPrefix !== null) {
                $tagName = $metadata->xmlRootPrefix . ':' . $metadata->xmlRootName;
            }
```

File: src/SoapMessageEventSubscriber.php [hunk 3]
```diff
@@ -56,9 +56,9 @@ public function addMessage(PreSerializeEvent $event)
                 throw new RuntimeException('Missing XmlRootName or XmlRootNamespace for ' . $event->getType()['name']);
             }
 
-            if($metadata->xmlRootPrefix !== null){
+            if ($metadata->xmlRootPrefix !== null) {
                 $tagName = $metadata->xmlRootPrefix . ':' . $metadata->xmlRootName;
-            }else{
+            } else {
```
Comment (proggeler) [inline src/SoapMessageEventSubscriber.php:61]: this else can be removed by moving the assignment before the if. (see SoapHeaderEventSubscriber.php)

File: tests/SoapHeaderEventSubscriberTest.php [hunk 4]
```diff
@@ -82,4 +83,63 @@ function (HandlerRegistry $registry) {
...
+        static::assertSame('Body', $xml->xpath('/*[local-name()="Envelope"]/*')[1]->getName());
+        static::assertCount(2, $xml->xpath('//*[local-name()="Languages"]/*[local-name()="language"]'));
+        static::assertCount(1, $xml->xpath('//*[local-name()="Languages"]/*[local-name()="count"]'));
+
+        $header = $xml->xpath('//*[local-name()="Header"]/*')[0]->children('http://xmpl-namespace.nl');
+        static::assertContains('http://xmpl-namespace.nl', $header->getNamespaces());
+        static::assertContains('ns', array_keys($header->getNamespaces()));
+        static::assertSame('dummy', strval($header->username));
+        static::assertSame('secret123!', strval($header->password));
+
```
[the remark this answers is not in the document]
Comment (Seriyyy95) [reply tests/SoapHeaderEventSubscriberTest.php]: Fixed

**Comment by Seriyyy95** [comment]
Fixed all of this.

**Review by proggeler — APPROVED** [review]

**Pull request merged**